mcaf/provider.go

```diff
@@ -4,22 +4,21 @@ package mcaf
 import (
 	"fmt"
 
-	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
-	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-// Provider returns a terraform.ResourceProvider.
-func Provider() terraform.ResourceProvider {
+// New returns a schema.Provider.
+func New() *schema.Provider {
```

Comment:
Is this a new best practice? While in Go term this is a good change, in TF terms this was the standard way to name this function.

Reply:
Seems so, noticed the change when checking to out the [terraform-provider-scaffolding](https://github.com/hashicorp/terraform-provider-scaffolding) project. Felt like a good time to align with that.

Wasn't so sure on moving `provider` to `internal/provider` though.

Reply:
Check... Well I should move it to `internal` as well. That is a good pattern to adopt I think...